test/libyul/evmCodeTransform/eof_dup256.yul

```diff
@@ -0,0 +1,2594 @@
+//     /* "":8339:8356   */
+//   calldataload
+//     /* "":8390:8393   */
+//   0xe1
+//     /* "":8377:8394   */
+//   calldataload
+//     /* "":8428:8431   */
+//   0xe2
+//     /* "":8415:8432   */
+//   calldataload
+//     /* "":8466:8469   */
+//   0xe3
+//     /* "":8453:8470   */
+//   calldataload
+//     /* "":8504:8507   */
+//   0xe4
+//     /* "":8491:8508   */
+//   calldataload
+//     /* "":8542:8545   */
+//   0xe5
+//     /* "":8529:8546   */
+//   calldataload
+//     /* "":8580:8583   */
+//   0xe6
+//     /* "":8567:8584   */
+//   calldataload
+//     /* "":8618:8621   */
+//   0xe7
+//     /* "":8605:8622   */
+//   calldataload
+//     /* "":8656:8659   */
+//   0xe8
+//     /* "":8643:8660   */
+//   calldataload
+//     /* "":8694:8697   */
+//   0xe9
+//     /* "":8681:8698   */
+//   calldataload
+//     /* "":8732:8735   */
+//   0xea
+//     /* "":8719:8736   */
+//   calldataload
+//     /* "":8770:8773   */
+//   0xeb
+//     /* "":8757:8774   */
+//   calldataload
+//     /* "":8808:8811   */
+//   0xec
+//     /* "":8795:8812   */
+//   calldataload
+//     /* "":8846:8849   */
+//   0xed
+//     /* "":8833:8850   */
+//   calldataload
+//     /* "":8884:8887   */
+//   0xee
+//     /* "":8871:8888   */
+//   calldataload
+//     /* "":8922:8925   */
+//   0xef
+//     /* "":8909:8926   */
+//   calldataload
+//     /* "":8960:8963   */
+//   0xf0
+//     /* "":8947:8964   */
+//   calldataload
+//     /* "":8998:9001   */
+//   0xf1
+//     /* "":8985:9002   */
+//   calldataload
+//     /* "":9036:9039   */
+//   0xf2
+//     /* "":9023:9040   */
+//   calldataload
+//     /* "":9074:9077   */
+//   0xf3
+//     /* "":9061:9078   */
+//   calldataload
+//     /* "":9112:9115   */
+//   0xf4
+//     /* "":9099:9116   */
+//   calldataload
+//     /* "":9150:9153   */
+//   0xf5
+//     /* "":9137:9154   */
+//   calldataload
+//     /* "":9188:9191   */
+//   0xf6
+//     /* "":9175:9192   */
+//   calldataload
+//     /* "":9226:9229   */
+//   0xf7
+//     /* "":9213:9230   */
+//   calldataload
+//     /* "":9264:9267   */
+//   0xf8
+//     /* "":9251:9268   */
+//   calldataload
+//     /* "":9302:9305   */
+//   0xf9
+//     /* "":9289:9306   */
+//   calldataload
+//     /* "":9340:9343   */
+//   0xfa
+//     /* "":9327:9344   */
+//   calldataload
+//     /* "":9378:9381   */
+//   0xfb
+//     /* "":9365:9382   */
+//   calldataload
+//     /* "":9416:9419   */
+//   0xfc
+//     /* "":9403:9420   */
+//   calldataload
+//     /* "":9454:9457   */
+//   0xfd
+//     /* "":9441:9458   */
+//   calldataload
+//     /* "":9492:9495   */
+//   0xfe
+//     /* "":9479:9496   */
+//   calldataload
+//     /* "":9530:9533   */
+//   0xff
+//     /* "":9517:9534   */
+//   calldataload
+//     /* "":9559:9562   */
+//   0x0100
+//     /* "":9546:9563   */
+//   calldataload
+//     /* "":9543:9579   */
+//   rjumpi{tag_1}
+//     /* "":25:16045   */
+// tag_2:
+//     /* "":9588:9603   */
+//   dupn{256}
```

Comment:
The conditional jump splits basic blocks and prevents us from moving the ``dupn{256}`` further up (as would currently happen otherwise, see https://github.com/ethereum/solidity/pull/15877).
Note that due to the branching, the test is noticeably slow, since the current logic for combining stacks is slow in the number of slots to consider.

Reply:
This comment and the others you added in tests are very useful. Could you put them in the tests themselves?

After having to review the EOF PRs that bulk update tons of tests I can say that determining what exactly is being tested, what is expected and which parts are essential and which are not is really time consuming even for someone who knows compiler internals. For contributors it may be outright impossible. Our test names are too vague to be useful and names used inside don't help either. We should in general have more comments like this inside tests.

Reply:
Added some comments.

Reply:
Nice. Though TBH I wouldn't mind even more of the details that were in the original comments :)